### PR TITLE
Fixes & Ionizers for 0.3.0-rc3

### PR DESCRIPTION
**Pull request by ax3l**
Adds the following changes to the `release-0.3.0` branch:

**Features:**
- Multiple ionizers per species:
  `ionizer< ... >` became a sequence of `ionizers< ... >` #1999

**Fixes:**
- Thomas-Fermi:
  - fixes in domain decomposition #2007
  - fixes compile issue (missing include) #2003
- BSI models restructured #2013
- ADK: fix effective principal quantum number `nEff` #2011
- multiple ionization algorithms can be applied per species,
  e.g. cut-off barrier suppression ionization (BSI),
  probabilistic field ionization (ADK) and collisional ionization #1999
- FieldTmp: Gather support to fill GUARD #2009
- Fix undefined device variable in tilted laser pulse #2017
- CompileTime Accessor: Type (::type) #1998
- TBG Macros: Outdated Comment #2004

Increases the state of the `0.3.0` release candidate to `rc3`.

### CODE REVIEW

**Comment by ax3l** [comment]
thx, typos fixed

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,7 +23,9 @@ streamlined to use Sphinx from now on.
  - use C++11 `using` instead of `typedef`
  - removed `Config` suffix in file names #1965
  - `gasConfig` is now `density`
- - `speciesDefinition`: simplified `Particles<>` interface #1711 #1942
+ - `speciesDefinition`:
+   - simplified `Particles<>` interface #1711 #1942
+   - `ionizer< ... >` became a sequence of `ionizers< ... >` #1999
```
Comment (PrometheusPi) [inline CHANGELOG.md]: QUESTION:
@ax3l This is not a bug fix but a feature - should this be back-ported? (Or is this required to add other fixes?)

File: src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp [hunk 2]
```diff
@@ -75,7 +75,11 @@ namespace ionization
 
         using FrameType =  typename SrcSpecies::FrameType;
 
-        /* specify field to particle interpolation scheme */
+        /** specify field to particle interpolation scheme
+         *
+         * @todo this needs to be done independently/twice if ion species (rho) and electron
+         *       species (ene) are of different shape
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp]: QUESTION:
@n01r Why is the ion species called rho and the electron species ene?
Where do you use these abbreviations?

File: CHANGELOG.md (hunk 1 above)
Comment (ax3l) [inline CHANGELOG.md]: offline: it's a feature we want included in 0.3.0 and we had to branch early this time. usually, we don't add features to already open release candidates, only bug fixes

File: src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp (hunk 2 above)
Comment (ax3l) [inline src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp]: out of scope here, he just writes that he takes the density of the ion species and the avg energy ("temperature") from the electrons

**Pull request merged**